### PR TITLE
Mdranges

### PR DESCRIPTION
**Pull request by tresreid**
Move MD range creation function to GPU

### CODE REVIEW

**Comment by tresreid** [comment]
Timing:
image

Validation: https://www.classe.cornell.edu/~mgr85/www/SegmentLinking_keep/PR274/minidoubletRanges_PU200_NEVT-1_069c78D-PU200/summary/
Comparison: https://www.classe.cornell.edu/~mgr85/www/SegmentLinking_keep/PR274/PR274comparison_398270-PU200_069c78D-PU200/summary/

**Comment by VourMa** [comment]
I see that this PR carries also the changes from PR #273. Should we merge that one first, so that the duplicate changes hopefully disappear? And, if not, rebase this one to the new master to clean it up?

**Comment by VourMa** [comment]
Unfortunately, even after merging PR #273, the duplicate changes still appear here. Could you clean this PR up by rebasing?

**Comment by tresreid** [comment]
duplicates are gone.

File: SDL/Event.cu
Comment (VourMa) [inline SDL/Event.cu:690]: Fix indentation?

File: SDL/Event.cu
Comment (VourMa) [inline SDL/Event.cu:912]: Fix indentation?

**Review by VourMa — COMMENTED** [review]
Just formatting comments on my side.

By the way, out of curiosity, what was special in the MD case that made it more complicated? It wasn't obvious to me from the code changes.

**Comment by tresreid** [comment]
With suggestions:
image

Compared to master before PR272 (running with same conditions):
image

So it seems like the 8 streams is running comparable to master though it is still about the same as 6 streams. I could argue that the cudaMalloc and more importantly the cudaFree introduced has a blocking effect across the device and not the stream. And with multiple streams, when fully pipelined, this happens more often. So Gavin's suggestion of using the caching allocator might make this run more smoothly. But this could also just be a random fluctuation.

**Review by VourMa — APPROVED** [review]
The PR looks good after the updates, thanks! There are a couple of comments/weird indentations but I don't think it makes sense to delay merging this any longer, especially since this code is going to be touched again soon. I am approving and @GNiendorf can click the merge button, if he's also content with the updates.

**Comment by GNiendorf** [comment]
Addresses my two concerns and runs fine, so I'm merging this.

**Pull request merged**
